cdp-agentkit-core/typescript/src/farcaster_agentkit.ts

```diff
@@ -0,0 +1,101 @@
+import { z } from "zod";
+import { FarcasterAction, FarcasterActionSchemaAny } from "./actions/cdp/social/farcaster";
+
+/**
+ * Schema for the options required to initialize the FarcasterAgentkit.
+ */
+export const FarcasterAgentkitOptions = z
+  .object({
+    apiKey: z
+      .string()
+      .min(1, "The Neynar API key for Farcaster is required")
+      .describe("The Neynar API key for Farcaster"),
+    managedSigner: z
+      .string()
+      .min(1, "The Neynar Managed Signer UUID for Farcaster is required")
+      .describe("The Neynar Managed Signer UUID for Farcaster"),
+  })
+  .strip()
+  .describe("Options for initializing FarcasterAgentkit");
+
+/**
+ * Schema for the environment variables required for FarcasterAgentkit.
+ */
+const EnvSchema = z.object({
+  NEYNAR_API_KEY: z
+    .string()
+    .min(1, "NEYNAR_API_KEY is required")
+    .describe("The Neynar API key for Farcaster"),
+  NEYNAR_MANAGED_SIGNER: z
+    .string()
+    .min(1, "NEYNAR_MANAGED_SIGNER is required")
+    .describe("The Neynar Managed Signer UUID for Farcaster"),
```

Comment:
How are these configs injected into the action during run?

Reply:
They're not injected into the action (it's not needed, as it's an environment config), but just enforced and validated upon FarcasterAgentkit construction.

Reply:
Per conversation in Telegram, going to follow up with options-capability, using env variables for now. Will leave FarcasterAgentkitOptions since it's not going to be the case that the action will always use env vars.